nextstrain-pathogen.yaml

```diff
@@ -8,3 +8,22 @@
 ---
 compatibility:
   nextstrain run: true
+# [Discussion] Realistically we won't implement buildpacks for a number of months
+# and so that can't block our dependency checking. We can try to second guess the
+# correct syntax or (easier) deliberately choose a top-level key name which won't
+# be used by buildpacks and then migrate to the buildpacks syntax when we implement
+# them for each repo.
+dependencies: # let the bikeshedding begin!
+  # [TODO XXX] the dependencies listed here aren't actually what Zika needs, these
+  # are used to test out the `check_pathogen_required_versions` functionality
+  py:
+    nextstrain-augur: ">=30,!=31.3"
+    boto3: ">=1.38"
+    snakemake: ">=9,<10"
+  cli:
+    augur: '>30'
+    nextclade: '>=3.15'
+    this-program-doesnt-exist: '1'
+    nextstrain: '==11'
+    snakemake: "<=8"
+    aws: "==3"
```

Comment:
Does this need to differentiate between `py` and `cli` dependencies? I'm already confused by the `nextstrain-augur` vs `augur` dependencies listed...

Could the version check just try `importlib_version` first and if that fails, then try `subprocess.run`?

Reply:
> I'm already confused by the nextstrain-augur vs augur dependencies listed...

I've better documented the list in `nextstrain-pathogen.yaml` - the intention was for them to be a testing set rather than how an actual pathogen repo's dependencies would look like. 

> Could the version check just try importlib_version first and if that fails, then try subprocess.run?

Sure. That trades off explicitness for simplicity. I've adjusted the code accordingly.